tests/fourier/test_advanced_spectrum.py

```diff
@@ -0,0 +1,545 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the Fourier spectrum transform.
+"""
+from collections import OrderedDict
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane import numpy as pnp
+from pennylane.fourier.advanced_spectrum import advanced_spectrum, _process_ids
+from pennylane.transforms import classical_jacobian
+
+
+def circuit_0(a):
+    [qml.RX(a, wires=0) for i in range(4)]
```

Comment:
Yes because in a list comprehension the inner code is executed in any case.
Although I recently learned that `for i in range(4): qml.RX(a, wires=0)` (in one line) also is valid Python syntax.
Maybe that would be even clearer because no list is created this way (which then is discarded anyways)?

Reply:
I think it's fine as-is (plus it's just in the tests). As a side note seeing a for loop in one line like that is strangely unsettling :fearful:

Reply:
Haha yeah it is strange. `black` actually does not like it and breaks it up into 2 lines.